src/coreclr/jit/emit.cpp

```diff
@@ -3904,6 +3904,111 @@ void emitter::emitRecomputeIGoffsets()
 #endif
 }
 
+//----------------------------------------------------------------------------------------
+// emitDispCommentForHandle:
+//    Displays a comment for a handle, e.g. displays a raw string for GTF_ICON_STR_HDL
+//    or a class name for GTF_ICON_CLASS_HDL
+//
+// Arguments:
+//    handle - a constant value to display a comment for
+//    flags  - a flag that the describes the handle
+//
+void emitter::emitDispCommentForHandle(size_t handle, GenTreeFlags flag)
+{
+#ifdef DEBUG
+    if (handle == 0)
+    {
+        return;
+    }
+
+#ifdef TARGET_XARCH
+    const char* commentPrefix = "      ;";
+#else
+    const char* commentPrefix = "      //";
+#endif
+
+    flag &= GTF_ICON_HDL_MASK;
+    const char* str = nullptr;
+
+    if (flag == GTF_ICON_STR_HDL)
+    {
+        const WCHAR* wstr = emitComp->eeGetCPString(handle);
+        // NOTE: eGetCPString always returns nullptr on Linux/ARM
+        if (wstr == nullptr)
+        {
+            printf("%s string handle", commentPrefix);
+        }
+        else
+        {
+            const size_t actualLen = wcslen(wstr);
+            const size_t maxLength = 63;
```

Comment:
Curious why we have a limit for this? Why not just print the full method name?

Reply:
this specific limit is only for string literals, I don't think it's useful to print the whole strings, it's usually enough just to quickly distinguish them.

but the method name is also limited via %.64s - I'll remove it 👍